### PR TITLE
borrow History supports params

### PR DESCRIPTION
**Pull request by msaug**
## Summary

https://docs.ftx.com/#spot-margin endpoints should support start_time and end_time params.
Support is now available.

Added BorrowHistoryParams type, forwarded params to the requestWrapper

##  Checks

- [ ] Updated related documentation
- [ ] Updated/added related tests
- [X ] I have tested my changes locally

### CODE REVIEW

File: src/rest-client.ts [hunk 1]
```diff
@@ -376,8 +376,8 @@ export class RestClient {
     return this.requestWrapper.get(`spot_margin/market_info${suffix}`);
   }
 
-  getBorrowHistory(): GenericAPIResponse {
-    return this.requestWrapper.get(`spot_margin/borrow_history`);
+  getBorrowHistory(params:BorrowHistoryParams): GenericAPIResponse {
```
Comment (tiagosiebler) [inline src/rest-client.ts:379]: Thanks for contributing! You can actually use the `TimeRange` interface here:
```suggestion
  getBorrowHistory(params: TimeRange): GenericAPIResponse {
```

It's already used by a few endpoints and shares the same properties (both optional).

**Review by tiagosiebler — APPROVED** [review]
Thanks for the quick PR. There's an interface you can reuse here & some minor code style nitpicks.

**Comment by msaug** [comment]
Hey, thanks for the feedback. I should have paid attention ! 
Keep up the good work

**Review by tiagosiebler — APPROVED** [review]
Thanks & no worries, it's easy to miss 🙂

File: src/rest-client.ts [hunk 2]
```diff
@@ -376,8 +376,8 @@ export class RestClient {
     return this.requestWrapper.get(`spot_margin/market_info${suffix}`);
   }
 
-  getBorrowHistory(): GenericAPIResponse {
-    return this.requestWrapper.get(`spot_margin/borrow_history`);
+  getBorrowHistory(params: TimeRange): GenericAPIResponse {
```
Comment (tiagosiebler) [inline src/rest-client.ts:379]: Damn just after merging I realised something - the params are optional, so it should be `params?: TimeRange`. I'll do a quick fix with the release PR.

**Comment by tiagosiebler** [comment]
Released with fix in v1.1.7

File: src/rest-client.ts (hunk 2 above)
Comment (msaug) [reply src/rest-client.ts:379]: Ah yes indeed, pagination is always optional
So in this case it could be called without params...

Comment (tiagosiebler) [reply src/rest-client.ts:379]: Yes..totally missed it. If all properties in the interface are optional then the param should also be optional. I've updated it before publishing for 1.1.7, see #69

**Pull request merged**